### PR TITLE
Implemented Custom Actions condition-checking functions

### PR DESCRIPTION
**Pull request by brentd313**
I implemented a fundamental set of condition-checking functions for custom action execution. The implementations are located in the src/custom-actions/src directory in the file "custom-actions-cond.c", and there is a corresponding header file in src/custom-actions/include. I also wrote unit tests for each of the implemented functions in the tests/custom-actions directory in the "test_ca_cond.c" file. I created and modified the necessary CMakeLists.txt files in order for the implementations and the tests to compile. As of now, the code is compiling, all tests are passing, and these implementations should not affect any other team's code. Although we may want to expand or modify this set of condition-checking functions at some point, these will provide a lot of the basic condition-checking functionalities.

### CODE REVIEW

File: src/custom-actions/src/custom-actions-cond.c [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+
+/* See custom-actions-cond.h */
+int check_gte(attribute_t *a1, attribute_t *a2)
+{
+    num_comp(a1, a2, GTE);
+}
+
+
+/* To be implemented:
```
Comment (MaxineK36) [inline src/custom-actions/src/custom-actions-cond.c:224]: is there a backlogged issue for this?

File: tests/custom-actions/test_ca_cond.c [hunk 2]
```diff
@@ -0,0 +1,316 @@
+#include <criterion/criterion.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include "../src/custom-actions/include/custom-actions-cond.h"
```
Comment (MaxineK36) [inline tests/custom-actions/test_ca_cond.c:4]: this should be able to be changed to `#include "custom-actions-cond.h`

File: tests/custom-actions/test_ca_cond.c [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
```
Comment (MaxineK36) [inline tests/custom-actions/test_ca_cond.c:5]: You should add a quick note explaining that this is a helper, and explaining why it's here

File: src/custom-actions/src/custom-actions-cond.c (hunk 1 above)
Comment (brentd313) [reply src/custom-actions/src/custom-actions-cond.c:224]: There are a couple condition and effect functions that are intimately involved with some of the more complex data structures of other teams. Would it be alright if I add a task to the backlog to flesh these out?

Comment (MaxineK36) [reply src/custom-actions/src/custom-actions-cond.c:224]: yep! let me know if you'd like me to look at the issue once you write it, and make sure to add labels for all the relevant teams.

File: tests/custom-actions/test_ca_cond.c (hunk 2 above)
Comment (brentd313) [reply tests/custom-actions/test_ca_cond.c:4]: Making this change causes the build to break. Could it be a problem with the CMake in the tests/custom-actions directory?

Comment (brentd313) [reply tests/custom-actions/test_ca_cond.c:4]: I got it fixed!

File: tests/custom-actions/test_ca_cond.c (hunk 3 above)
Comment (brentd313) [reply tests/custom-actions/test_ca_cond.c:6]: I added a note in the documentation of the new_attribute function, explaining its use in this test file.

**Comment by brentd313** [comment]
This pull request has been merged into the dev branch, and the custom-actions/conditions branch has been deleted. To review, the functions in this pull request provide basic ways to check conditions that will determine whether or not a custom action can be executed or not. In this pull request, I implemented these functions and wrote unit tests. While these provide a good set of tools for checking conditions of a function, we will continue to update these implementations with improvements or new functions as need for new functionality presents itself. The check_in() function in "custom-actions-cond.h" and discussed above provides a good example of what a more advanced condition-checking function might be for future implementation. The code included in this pull request can be found primarily in src/custom-actions and tests/custom-actions.

**Comment by MaxineK36** [comment]
Issue Score: ✔️++

Comments:
This was a great PR! The changes were incremental, and the description was very helpful as a guide for reviewers.

**Pull request merged**